x/ccv/provider/types/msg.go

```diff
@@ -28,23 +29,37 @@ const (
 )
 
 var (
-	_ sdk.Msg = &MsgAssignConsumerKey{}
-	_ sdk.Msg = &MsgSubmitConsumerMisbehaviour{}
-	_ sdk.Msg = &MsgSubmitConsumerDoubleVoting{}
-	_ sdk.Msg = &MsgOptIn{}
-	_ sdk.Msg = &MsgOptOut{}
-	_ sdk.Msg = &MsgSetConsumerCommissionRate{}
+	_ sdk.Msg = (*MsgAssignConsumerKey)(nil)
+	_ sdk.Msg = (*MsgConsumerAddition)(nil)
+	_ sdk.Msg = (*MsgConsumerRemoval)(nil)
```

Comment:
Should `MsgConsumerModification` be here as well? Similarly below in `sdk.HasValidateBasic`.